core/blockchain.go

```diff
@@ -993,6 +992,7 @@ func (bc *BlockChain) Stop() {
 		if snapBase, err = bc.snaps.Journal(bc.CurrentBlock().Root); err != nil {
 			log.Error("Failed to journal state snapshot", "err", err)
 		}
+		bc.snaps.Release()
```

Comment:
Traced this down to PR 28387, fixes a small memory leak in some testing cases, hope it improves our testing too.